inference_segmentation.py

```diff
@@ -437,10 +436,16 @@ def main(params: Union[DictConfig, dict]) -> None:
 
     # LOOP THROUGH LIST OF INPUT IMAGES
     for aoi in tqdm(list_aois, desc='Inferring from images', position=0, leave=True):
-        inference_image = working_folder / f"{aoi.raster_name.stem}_inference.tif"
-        inference_heatmap = working_folder / f"{aoi.raster_name.stem}_heatmap.tif"
-        temp_file = working_folder / f"{aoi.raster_name.stem}.dat"
-        logging.info(f'\nReading image: {aoi.raster_name.stem}')
+        inference_image = get_key_def(
```

Comment:
I dont know if looping on a `get_key_def` is the best solution, time wise I think it will be wiser to make the call outside the function and if `{aoi.aoi_id}_pred.tif` not there then reassign inference_image. 

But I can be wrong, its just that time wise calling a function is more time consuming that just checking if it there.